canonicals.js

```diff
@@ -0,0 +1,59 @@
+var APP = APP || {};
+(function(app) {
+
+    var canonicals = {
+
+        closures: [750486],
+        asynchronous: [14220321],
+        inheritance: [11072556, 7486825]
+
+    };
+
+    var ids = Object.keys(canonicals).map(function(category) {
+        return canonicals[category].join(';');
+    });
+
+    app.xhr.fetch(ids.join(';')).then(function success(questions) {
+        var canonicalQuestions = questions.items.reduce(function(questionMap, soQuestion) {
+            return Object.keys(canonicals).reduce(function(questionArr, category) {
+                questionMap[category] = questionMap[category] || [];
+                if (containsQuestionId(canonicals[category], soQuestion.question_id)) {
+                    questionMap[category].push(soQuestion);
+                }
+                return questionMap;
+            }, [])
+            return questionMap;
+        }, {});
+
```

Comment:
This looks terrible, there's a better way, right? There has to be. I feel like I've completely missed the point of using reduce.